WordPress/gutenberg
# Editor Settings: Rename the getPostLinkProps setting

## What?
Renames the `getPostLinkProps` and `goBack` editor setting to `onNavigateToEntityRecord`
`onNavigateToPreviousEntityRecord` and removes the href property from the return value in line with [feedback here](https://github.com/WordPress/gutenberg/pull/57700#discussion_r1464407249).

## Why?
The current naming doesn't work well for the wider usage of this setting outside the pattern focused editing it was originally added for. Also, the `href` prop does not work in the post editor so shouldn't be included.

## How?
Renamed all the relevant hooks, settings, and usages.

## Testing Instructions
- Add a new post and select the `Template - Edit Template` option in the right post panel
- Check that you navigate to the template and can edit it
- Use the back button in the top editor and make sure you go back to the post
- Check that when you click the post Save/Update button the template changes show as needing saving
- Also try the `Template - Create new template` option in right post panel and check it works
- Try the `Template - Edit template` option again and this time make sure the `Go back` option in the success snackbar works
- In the site editor open a page and repeat all of the above actions. The only difference will be that the `Go back` button will not appear in the notification, this is because the routing works differently in the site editor, and the top back link is more prominent
